### PR TITLE
Draftify e2e in anticipation of a spec release

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

**Comment by ara4n** [comment]
um, why? Just because we haven't published it doesn't mean we shouldn't have stubs from it in the spec. I am strongly against randomly removing it, as it suggests perceptually that we're giving up on it - which is very far from the case!!!

**Comment by illicitonion** [comment]
Right now it documents a bunch of API endpoints which we don't want to release, so shouldn't live on master.

I expect Mr VDH to revive this on a somewhat-long-lived branch, and restore it to master when we're happy with it.

**Comment by ara4n** [comment]
okay, but I'd like to make an exception for this one.  I am very happy for the endpoint details to be removed from the release, but I would like to keep it in the module matrix, and keep a placeholder module for it, and have a WIP section that links to the places where drafts and stuff is happening so we make it clear that E2E is very much on our radar and still a WIP.

File: specification/modules/instant_messaging.rst [hunk 1]
```diff
@@ -269,11 +269,8 @@ code of 400.
 Security considerations
 -----------------------
 
-Messages sent using this module are not encrypted. Messages can be encrypted
-using the `E2E module`_.
+Messages sent using this module are not encrypted.
```
Comment (ara4n) [inline specification/modules/instant_messaging.rst]: "are not encrypted, although end to end encryption is in development (see `E2E module`_)"

**Comment by ara4n** [comment]
lgtm other than the clarification above. ftr the reason it went straight into the spec was to avoid it rotting to death in drafts, but i'm happy to keep the spec cleaner of WIP stuff.

File: specification/modules/instant_messaging.rst (hunk 1 above)
Comment (illicitonion) [inline specification/modules/instant_messaging.rst]: Done

File: specification/feature_profiles.rst [hunk 2]
```diff
@@ -25,13 +25,11 @@ Summary
  `VoIP`_                               Required   Required   Required   Optional   Optional
  `Content Repository`_                 Required   Required   Required   Optional   Optional
  `Managing History Visibility`_        Required   Required   Required   Required   Optional
- `End-to-End Encryption`_              Optional   Optional   Optional   Optional   Optional
```
Comment (richvdh) [inline specification/feature_profiles.rst]: shouldn't we keep this bit?

**Comment by richvdh** [comment]
This flies in the face of what we discussed at lunch yesterday about putting WIP things into branches. OTOH this LGTM for now, modulo the profile table.

File: specification/feature_profiles.rst (hunk 2 above)
Comment (illicitonion) [inline specification/feature_profiles.rst]: No; we shouldn't mandate to people that they do or don't implement something we haven't actually specified.

(I convinced Matthew of this yesterday, too :))

**Comment by illicitonion** [comment]
I am very happy for you to commit something in the future which takes it out of drafts and puts it onto a branch :)

**Pull request merged**